test/suites/integration/storage_test.go

```diff
@@ -16,104 +16,212 @@ package integration_test
+			pvc := test.PersistentVolumeClaim(test.PersistentVolumeClaimOptions{
+				VolumeName:       "test-volume",
+				StorageClassName: lo.ToPtr("non-existent-storage-class"),
+			})
+			pv := test.PersistentVolume(test.PersistentVolumeOptions{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: pvc.Spec.VolumeName,
+				},
+				StorageClassName: "non-existent-storage-class",
+			})
+			pod := test.Pod(test.PodOptions{
+				PersistentVolumeClaims: []string{pvc.Name},
+			})
+			env.ExpectCreated(nodeClass, nodePool, pv, pvc, pod)
+			env.EventuallyExpectHealthy(pod)
+			env.ExpectCreatedNodeCount("==", 1)
 		})
-
-		pod := test.Pod(test.PodOptions{
-			PersistentVolumeClaims: []string{pvc.Name},
+		It("should run a pod with a pre-bound persistent volume while respecting topology constraints", func() {
+			subnets := env.GetSubnets(map[string]string{"karpenter.sh/discovery": env.ClusterName})
+			shuffledAZs := lo.Shuffle(lo.Keys(subnets))
+
+			pvc := test.PersistentVolumeClaim(test.PersistentVolumeClaimOptions{
+				StorageClassName: lo.ToPtr("non-existent-storage-class"),
+			})
+			pv := test.PersistentVolume(test.PersistentVolumeOptions{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: pvc.Spec.VolumeName,
+				},
+				StorageClassName: "non-existent-storage-class",
+				Zones:            []string{shuffledAZs[0]},
+			})
+			pod := test.Pod(test.PodOptions{
+				PersistentVolumeClaims: []string{pvc.Name},
+			})
+			env.ExpectCreated(nodeClass, nodePool, pv, pvc, pod)
+			env.EventuallyExpectHealthy(pod)
+			env.ExpectCreatedNodeCount("==", 1)
 		})
-
-		env.ExpectCreated(nodeClass, nodePool, sc, pvc, pod)
-		env.EventuallyExpectHealthy(pod)
-		env.ExpectCreatedNodeCount("==", 1)
-		env.ExpectDeleted(pod)
 	})
-})
 
-var _ = Describe("Static PVC", func() {
-	It("should run a pod with a static persistent volume", func() {
-		storageClassName := "nfs-test"
-		bindMode := storagev1.VolumeBindingWaitForFirstConsumer
-		sc := test.StorageClass(test.StorageClassOptions{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: storageClassName,
-			},
-			VolumeBindingMode: &bindMode,
+	Context("Dynamic", func() {
+		var storageClass *storagev1.StorageClass
+		BeforeEach(func() {
+			// Ensure that the EBS driver is installed, or we can't run the test.
+			var ds appsv1.DaemonSet
+			if err := env.Client.Get(env.Context, client.ObjectKey{
+				Namespace: "kube-system",
+				Name:      "ebs-csi-node",
+			}, &ds); err != nil {
+				if errors.IsNotFound(err) {
+					Skip(fmt.Sprintf("skipping dynamic PVC test due to missing EBS driver %s", err))
+				} else {
+					Fail(fmt.Sprintf("determining EBS driver status, %s", err))
+				}
+			}
+			storageClass = test.StorageClass(test.StorageClassOptions{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-storage-class",
+				},
+				Provisioner:       aws.String("ebs.csi.aws.com"),
+				VolumeBindingMode: lo.ToPtr(storagev1.VolumeBindingWaitForFirstConsumer),
+			})
+			ExpectSetEBSDriverLimit(1)
```

Comment:
Just curious: Does this need to be set in each test case or just in the ones where we care about the volume limit?